### PR TITLE
skopeo: 0.2.0 -> 1.0.0, don't set policy and tmpdir during build

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nlewo** [comment]
I think Skopeo should work out of the box without having to specify any specific arguments, excepting for security reasons.
We should instead work on Skopeo to improve its current configuration file management: they are working hard on rootless stuffs but a root account is required to use Skopeo out-of-the-box :/

**Comment by zowoq** [comment]
> I think Skopeo should work out of the box without having to specify any specific arguments, excepting for security reasons.

I think `skopeo` disregarding the NixOS managed `/etc/containers/policy.json` without communicating that to the user could perhapts be called a security issue.

> We should instead work on Skopeo to improve its current configuration file management: they are working hard on rootless stuffs but a root account is required to use Skopeo out-of-the-box

I agree with this, if they change this upstream in a consistent way for all of the tools that would be good. 

However I see this as a inconsistency that we should resolve as we are the ones that are setting the policy for at build time causing `/etc/containers/policy.json` to be ignored by `skopeo` while the other tools respect it and actually require it to function.

I think we should just accept that it doesn't work out of the box.

**Comment by zowoq** [comment]
Not necessarily a good alternative but we could the same method to set the default policy for _all_ of the tools and remove the `etc/containers/policy.json` managed by the module?

**Comment by zowoq** [comment]
@NixOS/podman Any thoughts on this?

File: pkgs/build-support/docker/default.nix [hunk 1]
```diff
@@ -95,7 +95,7 @@ rec {
       sourceURL = "docker://${imageName}@${imageDigest}";
       destNameTag = "${finalImageName}:${finalImageTag}";
     } ''
-      skopeo --override-os ${os} --override-arch ${arch} copy "$sourceURL" "docker-archive://$out:$destNameTag"
+      skopeo --insecure-policy --tmpdir=/tmp --override-os ${os} --override-arch ${arch} copy "$sourceURL" "docker-archive://$out:$destNameTag"
```
Comment (Mic92) [inline pkgs/build-support/docker/default.nix:98]: How about ?

```suggestion
      skopeo --insecure-policy --tmpdir=$TMPDIR --override-os ${os} --override-arch ${arch} copy "$sourceURL" "docker-archive://$out:$destNameTag"
```

File: pkgs/build-support/docker/nix-prefetch-docker [hunk 2]
```diff
@@ -12,6 +12,7 @@ finalImageTag=
 hashType=$NIX_HASH_ALGO
 hashFormat=$hashFormat
 format=nix
+skopeoCmd="skopeo --insecure-policy --tmpdir=/tmp"
```
Comment (Mic92) [inline pkgs/build-support/docker/nix-prefetch-docker:15]: see above

**Comment by adisbladis** [comment]
> We should instead work on Skopeo to improve its current configuration file management: they are working hard on rootless stuffs but a root account is required to use Skopeo out-of-the-box

This is a very good longer term goal, but in the mean time setting the policy to a built-in causing Skopeo to ignore `/etc/containers/policy.json` is the wrong call and completely inconsistent with the rest of the libpod/libcontainer ecosystem.

This issue is about more than just Skopeo, and more than just this one file. See https://github.com/containers/libpod/issues/6053.

**Comment by nlewo** [comment]
> I think we should just accept that it doesn't work out of the box.

It seems you are right :(

> This issue is about more than just Skopeo, and more than just this one file. See containers/libpod#6053.

I don't think the Graham proposal would fix our current issue.

**Comment by nlewo** [comment]
@zowoq Could you rebase, `nixosTests.docker-tools` is fixed on master.

**Comment by zowoq** [comment]
Bumped to [1.0.0](https://github.com/containers/skopeo/releases/tag/v1.0.0).

**Comment by zowoq** [comment]
This broke `nix-prefetch-docker`, I've opened https://github.com/NixOS/nixpkgs/pull/88856 to fix it.

**Pull request merged**